### PR TITLE
MFA

### PR DESCRIPTION
**Pull request by louischan-oursky**
connect #522

### CODE REVIEW

**Comment by louischan-oursky** [comment]
@kiootic In case you forgot this PR :(

File: packages/skygear-core/src/error.ts [hunk 1]
```diff
@@ -1,26 +1,95 @@
 import { JSONObject } from "./types";
 
+/**
+ * @public
+ */
+export const SkygearErrorNameUnexpectedError = "UnexpectedError" as const;
```
Comment (kiootic) [inline packages/skygear-core/src/error.ts]: Can we consolidate the error names in an object? e.g.
```typescript
export const SkygearErrorNames = {
    UnexpectedError: 'UnexpectedError',
    // ...
};
```

File: packages/skygear-core/src/container.ts [hunk 2]
```diff
@@ -187,8 +187,18 @@ export class AuthContainer<T extends BaseAPIClient> {
...
+    // the access token in the request is invalid.
+    //
+    // If the request does not have access token at all,
+    // the server simply returns NotAuthenticated error without the header.
+    //
+    // Therefore, we have to keep the invalid token so that
+    // if refresh fails due to other reasons, the whole process
+    // can be retried.
+    //
+    // await this.parent.storage.delAccessToken(this.parent.name);
```
[the remark this answers is not in the document]
Comment (louischan-oursky) [reply packages/skygear-core/src/container.ts]: I kept it deliberately to warn us that they should not be there.

File: packages/skygear-core/src/error.ts (hunk 1 above)
Comment (louischan-oursky) [reply packages/skygear-core/src/error.ts]: OK.

**Comment by louischan-oursky** [comment]
@kiootic Updated.

**Pull request merged**